### PR TITLE
Fix reuploading problemset.

### PR DESCRIPTION
**Pull request by nickygerritsen**
Also fix uploading using PUT and some small bugs in the contest import script wrt problemset upload.

### CODE REVIEW

File: webapp/src/EventListener/ParseFilesForPutListener.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+    public function __invoke(ControllerEvent $event): void
+    {
+        // When we have a PUT request, the files on the request will not be filled, since PHP doesn't
+        // fill the $_FILES array. We can manually parse the request to fill it using a library.
+        // This code is based on https://github.com/symfony/symfony/issues/36409#issuecomment-612442260.
+        $request = $event->getRequest();
+        if ($request->isMethod('PUT')) {
+            $document = HttpFoundation::convert($request);
+            if (!$document->isMultiPart()) {
```
Comment (meisterT) [inline webapp/src/EventListener/ParseFilesForPutListener.php:21]: what is happening here?

Comment (nickygerritsen) [reply webapp/src/EventListener/ParseFilesForPutListener.php:21]: On that line or the whole file? The whole file I tried to explain in the comment and that line... checks if this request was actually a multipart upload, which is how browsers / HTTP do file uploads.

Comment (meisterT) [reply webapp/src/EventListener/ParseFilesForPutListener.php:21]: This line. When can this be a non-multipart request but end up here? And should this silently return in this case?

Comment (nickygerritsen) [reply webapp/src/EventListener/ParseFilesForPutListener.php:21]: If you do a JSON PUT for example? I will add a comment.

File: webapp/tests/Unit/Controller/API/ContestControllerAdminTest.php [hunk 2]
```diff
@@ -200,6 +200,23 @@ public function testProblemsetManagement(): void
         $callbackData = ob_get_clean();
         self::assertEquals(file_get_contents($problemsetFile), $callbackData);
 
+        // Upload the problemset again, this time using PUT to also test that
+        $problemsetFile = __DIR__ . '/../../../../../webapp/public/doc/logos/DOMjudgelogo.pdf';
+        $problemset = new UploadedFile($problemsetFile, 'DOMjudgelogo.pdf');
+        $this->verifyApiJsonResponse('PUT', $url . '/problemset', 204, $this->apiUser, null, ['problemset' => $problemset]);
+
+        // Verify we still have a problemset
+        $object = $this->verifyApiJsonResponse('GET', $url, 200, $this->apiUser);;
```
Comment (vmcj) [inline webapp/tests/Unit/Controller/API/ContestControllerAdminTest.php:209]:
```suggestion
        $object = $this->verifyApiJsonResponse('GET', $url, 200, $this->apiUser);
```

File: webapp/src/EventListener/ParseFilesForPutListener.php (hunk 1 above)
Comment (nickygerritsen) [reply webapp/src/EventListener/ParseFilesForPutListener.php:23]: Does this added comment make it more clear?

**Review by mpsijm — APPROVED** [review]
Can confirm that (re-)uploading the PDF with `POST` (or `PUT`) works on my machine with https://github.com/RagnarGrootKoerkamp/BAPCtools/commit/b94d81efe7a8a0a5b396e8be080ced367a153d2c :smile: 

(I now have trouble with reuploading problem ZIPs though, but I'll ~~#blamenicky~~ ask Nicky about that separately :stuck_out_tongue:)

**Review by meisterT — APPROVED** [review]

**Pull request merged**